apps/www/content/docs/dark-mode/vite.mdx

```diff
@@ -0,0 +1,151 @@
+---
+title: Vite
+description: Adding dark mode to your vite app.
+---
+
+## Dark mode
+
+<Steps>
+
+### Create a theme provider
+
+```tsx title="components/theme-provider.tsx"
+import { createContext, useContext, useEffect, useState } from "react"
+
+type ThemeProviderProps = {
+  children: React.ReactNode
+  defaultTheme?: string
```

Comment:
I would also like to see a better typing for Theme, but since you are getting stuff from localStorage, you probably have to use a parser like zod (already being used for the form component, so it isn't necessarily an extra dependency). In my own app I've done it like this:

```ts
import { z } from "zod"

const ThemeSchema = z.enum(["light", "dark", "system"]).default("system")

export type Theme = z.infer<typeof ThemeSchema>
```

```ts
type ThemeProviderProps = {
  children: React.ReactNode
  defaultTheme?: Theme
  storageKey?: string
}

type ThemeProviderState = {
  theme: Theme
  setTheme: (theme: Theme) => void
}

const initialState = {
  theme: "system" as Theme,
  setTheme: () => null,
}
```

```ts
  const [theme, setTheme] = useState(() =>
    ThemeSchema.parse(localStorage.getItem(storageKey) || defaultTheme)
  )
```

```ts
  const value = {
    theme,
    setTheme: (theme: Theme) => {
      localStorage.setItem(storageKey, theme)
      setTheme(theme)
    },
  }
```

Lmk what you think about that, maybe there is a better solution for it, but it definitely shouldn't be of type string imo.

Reply:
Zod could be good, but since this is a guide, I don't think we should be forcing the user to install a whole package to do that validation.

Reply:
I typed the theme as string having in mind user can handle different themes and not only dark/light mode necessary, instantiating them to saves whatever they want as theme